### PR TITLE
rsbkb: patch time crate for Rust 1.80

### PR DESCRIPTION
**Pull request by ProducerMatt**
## Description of changes

Update to dodge https://github.com/NixOS/nixpkgs/issues/332957

Also remove `rec` because `nix-tidyf` complained.

Result of `nixpkgs-review pr 333605` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rsbkb</li>
  </ul>
</details>

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/text/rsbkb/default.nix [hunk 1]
```diff
@@ -6,16 +6,19 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "rsbkb";
-  version = "1.4";
+  version = "1.4_823b35";
```
Comment (emilazy) [inline pkgs/tools/text/rsbkb/default.nix:9]: The Nixpkgs convention is `1.4-unstable-YYYY-MM-DD` with the commit date. (But if you move to `cargoPatches` that won’t be relevant.)

File: pkgs/tools/text/rsbkb/default.nix [hunk 2]
```diff
@@ -6,16 +6,19 @@
...
+    # NOTE: patched until next release for https://github.com/NixOS/nixpkgs/issues/332957
+    # owner = "trou";
+    owner = "ProducerMatt";
     repo = "rsbkb";
-    rev = "release-${version}";
-    hash = "sha256-c5+Q/y2tZfhXQIAs1W67/xfB+qz1Xn33tKXRGDAi3qs=";
+    # rev = "release-${version}";
+    rev = "823b35072d60d1526c5646bd850a39f056c3cfc7";
+    hash = "sha256-tc0nOx8Z8ytH0AS9nOhk/0ejBEKcg0lUF289TL1iRHM=";
   };
```
Comment (emilazy) [inline pkgs/tools/text/rsbkb/default.nix:19]: It may be better to keep the original `src` but add your diff to `cargoPatches`.

**Review by emilazy — CHANGES_REQUESTED** [review]
Build looks fine, thanks!

File: pkgs/tools/text/rsbkb/default.nix (hunk 2 above)
Comment (ProducerMatt) [reply pkgs/tools/text/rsbkb/default.nix:19]: Oh darn I missed this. In the docs, `cargoLock.lockFileContents` is presented first and `cargoPatches` is presented later in another section.

**Comment by ProducerMatt** [comment]
Was I correct to base the PR against `staging-next`?

**Review by emilazy — APPROVED** [review]
Result of `nixpkgs-review pr 333605` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rsbkb</li>
  </ul>
</details>

Thanks!

Targeting `staging-next` is fine, but `master` would also have been fine. If `staging-next` had already merged, we’d have to retarget it on `master`. For something like this I would generally base it on and target `master` but use `nixpkgs-review` to check that it works rebased on top of the `staging-next` cycle with the breaking change.

**Pull request merged**